File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    
    s = brown + yellow;
    let h = 0;
    for(let w = 1; w < s + 1; w++){
        if (s % w == 0){
            h = s / w;
            if(w >= h){
                if ((w-2) * (h-2) == yellow){
                    answer.push(w);
                    answer.push(h);}
            }
        }
    }
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int sum = brown + yellow;
        
        for(int h = 1; h <= sum; h++){
            int w = 0;
            if(sum % h == 0){
                // 1 12, 2 6, 3 4
                // s가 가로
                w = sum / h;
                if(w >= h && (w - 2) * (h - 2) == yellow) return new int[]{w, h};
                
            }
        }
        return answer;
    }
}